server/sonar-server/src/main/java/org/sonar/server/computation/step/QualityGateMeasuresStep.java

```diff
@@ -164,26 +166,39 @@ private void updateQualityGateStatusHolder(QualityGate qualityGate, QualityGateD
   }
 
   private void updateMeasures(Component project, Set<Condition> conditions, QualityGateDetailsDataBuilder builder) {
-    for (Condition condition : conditions) {
-      Optional<Measure> measure = measureRepository.getRawMeasure(project, condition.getMetric());
+    Multimap<Metric, Condition> conditionsPerMetric = from(conditions).index(ConditionToMetric.INSTANCE);
+    for (Map.Entry<Metric, Collection<Condition>> entry : conditionsPerMetric.asMap().entrySet()) {
+      Metric metric = entry.getKey();
+      Optional<Measure> measure = measureRepository.getRawMeasure(project, metric);
       if (!measure.isPresent()) {
         continue;
       }
 
-      EvaluationResult evaluationResult = new ConditionEvaluator().evaluate(condition, measure.get());
-
-      String text = evaluationResultTextConverter.asText(condition, evaluationResult);
+      MetricEvaluationResult metricEvaluationResult = evaluateQualityGate(measure.get(), entry.getValue());
+      String text = evaluationResultTextConverter.asText(metricEvaluationResult.condition, metricEvaluationResult.evaluationResult);
       builder.addLabel(text);
 
       Measure updatedMeasure = Measure.updatedMeasureBuilder(measure.get())
-        .setQualityGateStatus(new QualityGateStatus(evaluationResult.getLevel(), text))
+        .setQualityGateStatus(new QualityGateStatus(metricEvaluationResult.evaluationResult.getLevel(), text))
         .create();
-      measureRepository.update(project, condition.getMetric(), updatedMeasure);
+      measureRepository.update(project, metric, updatedMeasure);
 
-      builder.addEvaluatedCondition(condition, evaluationResult);
+      builder.addEvaluatedCondition(metricEvaluationResult);
     }
   }
 
+  private static MetricEvaluationResult evaluateQualityGate(Measure measure, Collection<Condition> conditions) {
```

Comment:
the `conditions` should be sorted in order to have deterministic results when multiple conditions (max 2 in theory) fail with the same severity. I suggest that condition on leak period takes precedence on the absolute condition.